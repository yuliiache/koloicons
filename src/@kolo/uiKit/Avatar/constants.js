export const AVATAR_SIZES = Object.freeze({
  WIDTH: 160,
  HEIGHT: 160,
});
