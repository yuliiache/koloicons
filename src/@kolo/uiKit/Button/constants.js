export const BUTTON_TEST_ID = 'buttonComponent';
export const BUTTON_TEXT_TEST_ID = `${BUTTON_TEST_ID}Text`;

export const BUTTON_COLOR = Object.freeze({
  PRIMARY: 'primary',
  ACCENT: 'accent',
  BASIC: 'basic',
});

export const BUTTON_SIZE = Object.freeze({
  SMALL: 'small',
  BIG: 'big',
  FULL_WIDTH: 'full-width',
});

export const BUTTON_TYPE = Object.freeze({
  BUTTON: 'button',
  SUBMIT: 'submit',
  RESET: 'reset',
});
