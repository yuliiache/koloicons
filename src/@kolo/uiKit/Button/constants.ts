export const BUTTON_TEST_ID = 'buttonComponent';
export const BUTTON_TEXT_TEST_ID = `${BUTTON_TEST_ID}Text`;

export enum ButtonColor {
  PRIMARY = 'primary',
  ACCENT = 'accent',
  BASIC = 'basic',
}

export enum ButtonSize {
  SMALL = 'small',
  BIG = 'big',
  FULL_WIDTH = 'full-width',
}

export enum ButtonType {
  BUTTON = 'button',
  SUBMIT = 'submit',
  RESET = 'reset',
}
