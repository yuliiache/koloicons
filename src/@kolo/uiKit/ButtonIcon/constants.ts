export const BUTTON_ICON_TEST_ID = 'buttonIconComponent';

export enum ButtonIconType {
  ROUND = 'round',
  SQUARE = 'square',
}
