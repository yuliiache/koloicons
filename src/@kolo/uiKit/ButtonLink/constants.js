export const BUTTONLINK_TEST_ID = 'buttonLinkComponent';

