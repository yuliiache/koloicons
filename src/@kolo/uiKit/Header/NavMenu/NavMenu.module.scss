.mainNav {
  display: flex;
  flex-direction: row;
}

.mainNavLinks {
  list-style: none;
  margin: 0;
  padding: 0;
  display: flex;
  flex-wrap: wrap;
  font-size: 20px;
  font-weight: 400;
}

.navLinksItem:not(:last-child) {
  margin-right: 24px;
}
