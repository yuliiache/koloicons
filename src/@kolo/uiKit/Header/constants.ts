export enum NavMenuItem {
  ICONS = 'Icons',
  PRICING = 'Pricing',
}

export enum HiddenPath {
  LOGIN = '/login',
  REGISTER = '/register',
  MY_ACCOUNT = '/my-account',
  COOKIE_POLICY = '/cookie-policy',
  PRIVACY_POLICY = '/privacy-policy',
  TERMS_OF_SERVICE = '/terms-of-service',
}
