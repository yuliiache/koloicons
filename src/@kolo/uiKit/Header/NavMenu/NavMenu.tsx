import {LinkType} from '@kolo/uiKit/Link/constants';
import Link from '@kolo/uiKit/Link/Link';
import {AppRoute} from 'constants/AppRoute';
import {FC, memo} from 'react';

import styles from './NavMenu.module.scss';

const NavMenu: FC = () => {
  return (
    <nav className={styles.mainNav}>
      <ul className={styles.mainNavLinks}>
        <li className={styles.navLinksItem}>
          <Link
            url={AppRoute.ICONS}
            type={LinkType.MAIN}
          >
            Icons
          </Link>
        </li>
        <li className={styles.navLinksItem}>
          <Link
            url={AppRoute.PRICING}
            type={LinkType.MAIN}
          >
            Pricing
          </Link>
        </li>
      </ul>
    </nav>
  );
};

export default memo(NavMenu);
