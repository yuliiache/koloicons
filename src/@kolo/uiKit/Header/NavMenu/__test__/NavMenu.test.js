import React from 'react';
import {BrowserRouter as Router} from 'react-router-dom';
import {cleanup, render} from '@testing-library/react';

import {NavMenuItem} from '../../constants';
import NavMenu from '../NavMenu';

describe('NavMenu component', () => {
  afterEach(cleanup);

  it('snapshot renders correctly', () => {
    const {asFragment} = render(
      <Router>
        <NavMenu />
      </Router>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('NavMenu renders correctly with NavMenuItem', () => {
    const {getByText} = render(
      <Router>
        <NavMenu />
      </Router>
    );

    Object.values(NavMenuItem).forEach((item) => {
      expect(getByText(item)).toBeInTheDocument();
    });
  });
});
