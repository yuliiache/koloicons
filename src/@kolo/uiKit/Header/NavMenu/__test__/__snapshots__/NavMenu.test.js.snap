// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavMenu component snapshot renders correctly 1`] = `
<DocumentFragment>
  <nav
    class="mainNav"
  >
    <ul
      class="mainNavLinks"
    >
      <li
        class="navLinksItem"
      >
        <a
          class="link link--main"
          data-testid="internalLinkComponent"
          href="/icons"
        >
          Icons
        </a>
      </li>
      <li
        class="navLinksItem"
      >
        <a
          class="link link--main"
          data-testid="internalLinkComponent"
          href="/pricing"
        >
          Pricing
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
